c/frame/replace.cc

```diff
@@ -0,0 +1,569 @@
+//------------------------------------------------------------------------------
+// Copyright 2018 H2O.ai
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//------------------------------------------------------------------------------
+#include "frame/py_frame.h"
+#include <type_traits>
+#include <unordered_set>
+#include "python/dict.h"
+#include "python/list.h"
+#include "utils/assert.h"
+#include "utils/progress.h"
+
+namespace py {
+
+
+PKArgs Frame::Type::args_replace(
+  2, 0, 0, false, false,
+  {"to_replace", "replace_with"},
+  "replace",
+R"(replace(self, replace_what, replace_with)
+--
+
+Replace given value(s) `replace_what` with `replace_with` in the entire Frame.
+
+For each replace value, this method operates only on columns of types
+appropriate for that value. For example, if `replace_what` is a list
+`[-1, math.inf, None, "??"]`, then the value `-1` will be replaced in integer
+columns only, `math.inf` only in real columns, `None` in columns of all types,
+and finally `"??"` only in string columns.
```

Comment:
Is it because you don't want to change the column type during replacement?

Reply:
If the user wanted to change a column's type, he can do it explicitly beforehand. Otherwise, the semantics is weird: suppose you're changing integer `99` into string `"--"`. What happens if the column doesn't have any 99s? Should it remain integer, or get converted into string anyways? Having an unpredictable result type makes it hard to reason about the code.

Reply:
I agree, also don't like an idea to change the column type during replacement as this seems to be completely separate functionality.